treenode.go

```diff
@@ -286,7 +286,11 @@ func (n *TreeNodeInstance) closeDispatch() error {
 		n.msgDispatchQueueWait <- true
 	}
 	n.msgDispatchQueueMutex.Unlock()
-	return n.ProtocolInstance().Shutdown()
+	pni := n.ProtocolInstance()
+	if pni == nil {
```

Comment:
That's something very disturbing as there should be no `TreeNodeInstance` living without having a `ProtocolInstance` attached to it.
Was there an error with it ? Was it only in the tests ?